tests/conftest.py

```diff
@@ -8164,3 +8167,29 @@ def finalizer():
         constants.ODF_MONITORING_TOOL_REPO, str(repo_dir), branch="main", tmp_repo=True
     )
     return repo_dir
+
+
+@pytest.fixture(scope="session")
+def setup_cnv(request):
+    """
+    Session scoped fixture to setup and cleanup CNV
+    based on need of the tests
+
+    """
+    cnv_obj = CNVInstaller()
+    installed = False
+    if not cnv_obj.post_install_verification():
+        cnv_obj.deploy_cnv(check_cnv_deployed=False, check_cnv_ready=False)
+        installed = True
+
+    def finalizer():
+        """
+        Clean up CNV deployment
+
+        """
+
+        # Uninstall CNV only if installed by this fixture
+        if installed:
+            cnv_obj.uninstall_cnv()
```

Comment:
Will it also clean the DataVolumes, imports, pv and all?

Reply:
this is not part of CNV clean up right? it should be part of cnv workload fixture and i think it will be taken care there